### PR TITLE
Update glibc-217 base image to use available base url

### PR DESCRIPTION
**Pull request by jbudz**
CentOS 7 is EOL.  Instead of using mirrorlist, this updates the base
image to use vault.

### CODE REVIEW

**Review by mistic — APPROVED** [review]
LGTM after revert the comments on the pipeline file

File: glibc-217/Dockerfile [hunk 1]
```diff
@@ -7,19 +7,27 @@ RUN groupadd --force --gid $GROUP_ID node
...
-         xz-utils \
-         devtoolset-9 \
-         glibc-devel
+        git \
+        curl \
+        make \
+        python2 \
+        python3 \
+        ccache \
+        xz \
```
Comment (jbudz) [inline glibc-217/Dockerfile:28]: why this change?

Comment (mistic) [reply glibc-217/Dockerfile]: ups good catch. That was just legacy from testing possible solutions until we get into this state. Just removed those changes @jbudz

File: glibc-217/Dockerfile [hunk 2]
```diff
@@ -7,8 +7,16 @@ RUN groupadd --force --gid $GROUP_ID node
 RUN adduser --gid $GROUP_ID --uid $USER_ID node
 
 RUN ulimit -n 1024 \
+    && sed -i -e 's!mirrorlist!#mirrorlist!g' /etc/yum.repos.d/CentOS-* \
+    && sed -i -e 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-* \
     && yum install -y epel-release \
     && yum install -y centos-release-scl-rh \
+    && sed -i -e 's!mirrorlist!#mirrorlist!g' /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo \
```
Comment (jbudz) [inline glibc-217/Dockerfile:14]: This doesn't appear to be needed

File: glibc-217/Dockerfile [hunk 3]
```diff
@@ -7,8 +7,16 @@ RUN groupadd --force --gid $GROUP_ID node
...
+    && if [ "$(uname -m)" == "x86_64" ]; then \
+      sed -i -e 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo; \
+    else \
+      sed -i -e 's|#baseurl=http://mirror.centos.org/centos|baseurl=http://vault.centos.org/altarch|g' /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo; \
```
Comment (jbudz) [inline glibc-217/Dockerfile:18]:
```suggestion
      sed -i -e 's|#baseurl=http://vault.centos.org/centos|baseurl=http://vault.centos.org/altarch|g' /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo; \
```

File: glibc-217/Dockerfile [hunk 4]
```diff
@@ -7,8 +7,16 @@ RUN groupadd --force --gid $GROUP_ID node
 RUN adduser --gid $GROUP_ID --uid $USER_ID node
 
 RUN ulimit -n 1024 \
+    && sed -i -e 's!mirrorlist!#mirrorlist!g' /etc/yum.repos.d/CentOS-* \
+    && sed -i -e 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-* \
     && yum install -y epel-release \
     && yum install -y centos-release-scl-rh \
+    && sed -i -e 's!mirrorlist!#mirrorlist!g' /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo \
+    && if [ "$(uname -m)" == "x86_64" ]; then \
+      sed -i -e 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo; \
```
Comment (jbudz) [inline glibc-217/Dockerfile:16]:
```suggestion
      sed -i -e 's|#baseurl=http://vault.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo; \
```

File: glibc-217/Dockerfile (hunk 2 above)
Comment (jbudz) [reply glibc-217/Dockerfile:14]: Ah, I see.  both are included now with vault for Sources.
